esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -889,6 +889,51 @@ void EsdlServiceImpl::handleFinalRequest(IEspContext &context,
                    "No target URL configured for %s!", mthdef.queryMethodName());
     }
 
+    try
+    {
+        Owned<IPTree> responseTree = createPTreeFromXMLString(out, ipt_ordered);
```

Comment:
Can you avoid loading the result into a ptree?  Seems they can get quite large.  

I think you could just find the right position in the string to 1. insert the new dataset into, or 2. overwrite adding the new dataset and then re-add the closing tags?